File: apps/web/src/data/dto/notification-settings-response.ts
export interface NotificationSettingsResponse {
	/** 일정 알림 받기 여부. */
	eventEnabled: boolean;
	/** 일정 알림 시점(예: "하루 전"). */
	eventReminder: string;
	/** 기념일 알림 받기 여부. */
	anniversaryEnabled: boolean;
	/** 기념일 알림 시점(예: "당일"). */
	anniversaryReminder: string;
	/** 상대방 활동 알림 받기 여부. */
	partnerActivityEnabled: boolean;
}

File: apps/api-kotlin/src/main/kotlin/com/couplecalendar/application/dto/response/NotificationSettingsResponse.kt
package com.couplecalendar.application.dto.response

import com.couplecalendar.domain.aggregate.NotificationSettings

data class NotificationSettingsResponse(
    val eventEnabled: Boolean,
    val eventReminder: String,
    val anniversaryEnabled: Boolean,
    val anniversaryReminder: String,
    val partnerActivityEnabled: Boolean
) {
    companion object {
        fun fromAggregate(settings: NotificationSettings): NotificationSettingsResponse =
            NotificationSettingsResponse(
                eventEnabled = settings.eventEnabled,
                eventReminder = settings.eventReminder,
                anniversaryEnabled = settings.anniversaryEnabled,
                anniversaryReminder = settings.anniversaryReminder,
                partnerActivityEnabled = settings.partnerActivityEnabled
            )
    }
}
